cayenne-crypto/src/main/java/org/apache/cayenne/crypto/reader/CryptoRowReaderFactoryDecorator.java

```diff
@@ -109,7 +113,17 @@ protected RowDescriptor encryptedRowDescriptor(RowDescriptor descriptor, Extende
             ExtendedType t = originalConverters[i];
 
             if (attribute != null && columnMapper.isEncrypted(attribute)) {
```

Comment:
Two serious issues with this code:

* It is called once per query. It will be *extremely* slow, as it does full DataMap scan
* It has side effect on the shared mapping. We are not supposed to change anything in the DataMap outside of map loader.